h/util/markdown.py

```diff
@@ -43,56 +63,6 @@ def _filter_link_attributes(_tag, name, value):
     return False
 
 
-MARKDOWN_ATTRIBUTES = {"a": _filter_link_attributes, "img": ["alt", "src", "title"]}
-
-ALLOWED_ATTRIBUTES = bleach.ALLOWED_ATTRIBUTES.copy()
-ALLOWED_ATTRIBUTES.update(MARKDOWN_ATTRIBUTES)
-
-
-class MathMarkdown(mistune.Markdown):
-    def output_block_math(self):
-        return self.renderer.block_math(self.token["text"])
-
-
-class MathInlineLexer(mistune.InlineLexer):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.rules.inline_math = re.compile(r"\\\((.*?)\\\)", re.DOTALL)
-        self.default_rules.insert(0, "inline_math")
-
-    def output_inline_math(self, match):
-        return self.renderer.inline_math(match.group(1))
-
-
-class MathBlockLexer(mistune.BlockLexer):
-    def __init__(self, *args, **kwargs):
-        super().__init__(*args, **kwargs)
-        self.rules.block_math = re.compile(r"^\$\$(.*?)\$\$", re.DOTALL)
-        self.default_rules.insert(0, "block_math")
-
-    def parse_block_math(self, match):
-        self.tokens.append({"type": "block_math", "text": match.group(1)})
-
-
-class MathRenderer(mistune.Renderer):
-    def block_math(self, text):  # pylint: disable=no-self-use
-        return f"<p>$${text}$$</p>\n"
-
-    def inline_math(self, text):  # pylint: disable=no-self-use
-        return f"\\({text}\\)"
-
-
-def render(text):
-    if text is not None:
-        return _sanitize(_get_markdown()(text))
-
-    return None
-
-
-def _sanitize(text):
-    return _get_cleaner().clean(text)
```

Comment:
I think it may have been extracted into a separate method so that it could be unit-tested directly. I recently rewrote the unit tests to go through `render()` instead (https://github.com/hypothesis/h/pull/7235)